### PR TITLE
Adding design doc for shared dependencies

### PR DESCRIPTION
**Pull request by mattfarina**

### CODE REVIEW

File: docs/shared-dependencies.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+When someone installs an application (e.g., another system service) that depends
+on Prometheus then Hypper should check to see if Prometheus is installed in the
+expected location. If not, it should attempt to install it. If Prometheus is
+already installed it would skip installing it and instead use that instance.
+
+## Technical Details
+
+This section contains the technical details on implementing the request. These
+are suggest and open for discussion.
```
Comment (viccuad) [inline docs/shared-dependencies.md:30]:
```suggestion
are suggestions and open for discussion.
```

File: docs/shared-dependencies.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+### Prometheus Example
+
+Consider the case where cluster operators want to have a single instance of
+Prometheus running as a system wide service. There should only be one instance
+running in the cluster and all (or most) applications that need prometheus
+would use this single instance.
+
+When someone installs an application (e.g., another system service) that depends
+on Prometheus then Hypper should check to see if Prometheus is installed in the
+expected location. If not, it should attempt to install it. If Prometheus is
```
Comment (kkaempf) [inline docs/shared-dependencies.md:24]: How would Hypper install a service as a 'system wide' service ?

File: docs/shared-dependencies.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+### Prometheus Example
+
+Consider the case where cluster operators want to have a single instance of
+Prometheus running as a system wide service. There should only be one instance
+running in the cluster and all (or most) applications that need prometheus
+would use this single instance.
+
+When someone installs an application (e.g., another system service) that depends
+on Prometheus then Hypper should check to see if Prometheus is installed in the
```
[the remark this answers is not in the document]
Comment (viccuad) [reply docs/shared-dependencies.md:23]: One possibility would be by convention, selecting a `hypper` namespace.

**Review by kkaempf — APPROVED** [review]

File: docs/shared-dependencies.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (mattfarina) [reply docs/shared-dependencies.md:23]: The namespace and release name are stored as annotations in the chart. If one application has a dependency on prometheus than we can fetch the prometheus chart and look at the annotations in the Chart.yaml file for the release name and namespace.

This is why a local cache would be useful. Less would need to be pulled over the network.

File: docs/shared-dependencies.md (hunk 2 above)
Comment (mattfarina) [reply docs/shared-dependencies.md:24]: Think of this as a cluster admin needing to install something just once at the cluster level. Like an operator, logging, etc.

**Review by viccuad — APPROVED** [review]

**Pull request merged**